test/runtests.jl

```diff
@@ -666,4 +666,36 @@ def try_call(f):
                        pybuiltin("Exception"))
 end
 
+@static if VERSION < v"0.7.0-DEV.5126" # julia#25261
+    # PyIterator not defined in this julia version
+else
+    @testset "PyIterator" begin
+        arr = [1,2]
+        o = PyObject(arr)
+        c_pyany = collect(PyCall.PyIterator{PyAny}(o))
+        @test c_pyany == arr
+        @test c_pyany[1] === arr[1]
+        @test c_pyany[2] === arr[2]
+
+        c_f64 = collect(PyCall.PyIterator{Float64}(o))
+        @test c_f64 == arr
+        @test eltype(c_f64) == Float64
+
+        i1 = PyObject([1])
+        i2 = PyObject([2])
+        l = PyObject([i1,i2])
+
+        piter = PyCall.PyIterator(l)
+        @test length(piter) == 2
+        @test length(collect(piter)) == 2
+        r1, r2 = collect(piter)
+        @test r1.o === i1.o
+        @test r2.o  === i2.o
+    end
+
+    # 594
+    @test collect(zip(py"iter([1, 2, 3])", 1:3)) == [(1, 1), (2, 2), (3, 3)]
```

Comment:
Let's add

```julia
@test collect(zip(PyCall.PyIterator{Int}(py"iter([1, 2, 3])"), 1:3)) == [(1, 1), (2, 2), (3, 3)]
```

to check `Base.IteratorSize(::Type{<: PyIterator})`.  Let's also add

```julia
@test collect(zip(PyCall.PyIterator(py"[1, 2, 3]"o), 1:3)) == [(1, 1), (2, 2), (3, 3)]
```

once introspection-based `Base.IteratorSize` is implemented.